build.gradle.kts

```diff
@@ -134,90 +140,113 @@ tasks.withType<JacocoReport> {
-        executionData(junitPlatformTest)
-        sourceSets(java.sourceSets["main"])
-        sourceDirectories = files(java.sourceSets["main"].allSource.srcDirs)
-        classDirectories = files(java.sourceSets["main"].output)
+
+tasks.withType<Test> {
+    // TODO: re-enable when TestFX (or the underlying JavaFX problem) is fixed
+    println("UI tests will not be run due to TestFX being broken when headless on Java 10.")
+    println("See: https://github.com/javafxports/openjdk-jfx/issues/66")
+    // Link: https://github.com/javafxports/openjdk-jfx/issues/66
+    useJUnitPlatform {
+        excludeTags("UI")
     }
 }
 
-/*
-     * Allows you to run the UI tests in headless mode by calling gradle with the -Pheadless argument
-     */
-if (project.hasProperty("jenkinsBuild") || project.hasProperty("headless")) {
-    println("Running UI Tests Headless")
-    junitPlatform {
-        filters {
-            tags {
-                /*
-                 * A category for UI tests that cannot run in headless mode, ie work properly with real windows
-                 * but not with the virtualized ones in headless mode.
-                 */
-                exclude("NonHeadlessTests")
-            }
-        }
+tasks.withType<Javadoc> {
+    isFailOnError = false
+}
+
+dependencies {
+}
+
+val nativeShadowTasks = NativePlatforms.values().map { platform ->
+    tasks.create<ShadowJar>("shadowJar-${platform.platformName}") {
+        classifier = platform.platformName
+        configurations = listOf(
+                project.configurations.compile,
+                project.configurations.getByName(platform.platformName)
+        )
+        from(
+                java.sourceSets["main"].output
+        )
     }
-    tasks {
-        "junitPlatformTest"(JavaExec::class) {
-            jvmArgs = listOf(
-                    "-Djava.awt.headless=true",
-                    "-Dtestfx.robot=glass",
-                    "-Dtestfx.headless=true",
-                    "-Dprism.order=sw",
-                    "-Dprism.text=t2k"
-            )
-        }
+}
+
+tasks.create("shadowJarAllPlatforms") {
+    nativeShadowTasks.forEach {
+        this.dependsOn(it)
     }
 }
 
+tasks.withType<ShadowJar>().configureEach {
+    exclude("module-info.class")
+}
+
+val sourceJar = task<Jar>("sourceJar") {
+    description = "Creates a JAR that contains the source code."
+    from(java.sourceSets["main"].allSource)
+    classifier = "sources"
+}
+
+val javadocJar = task<Jar>("javadocJar") {
+    dependsOn("javadoc")
+    description = "Creates a JAR that contains the javadocs."
+    from(java.docsDir)
+    classifier = "javadoc"
+}
+
 publishing {
     publications {
-        create<MavenPublication>("OutlineViewer") {
-            artifactId = "OutlineViewer"
-            getWPILibVersion()?.let { version = it }
-            val shadowJar: ShadowJar by tasks
-            artifact (shadowJar) {
-                classifier = null
+        create<MavenPublication>("app") {
```

Comment:
Not sure what difference this makes, if any

Reply:
The name there does not matter to what actually gets published